gl/src/lib.rs

```diff
@@ -305,18 +305,24 @@ impl Device for GLDevice {
     }
 
     fn get_vertex_attr(&self, program: &Self::Program, name: &str) -> GLVertexAttr {
-        let name = CString::new(format!("a{}", name)).unwrap();
-        let attr = unsafe {
-            gl::GetAttribLocation(program.gl_program, name.as_ptr() as *const GLchar) as GLuint
+        let cname = CString::new(format!("a{}", name)).unwrap();
+        let attr_int = unsafe {
+            gl::GetAttribLocation(program.gl_program, cname.as_ptr() as *const GLchar)
         }; ck();
-        GLVertexAttr { attr }
+        if attr_int == -1 {
+            panic!("Vertex attribute '{}' does not exist in program", name);
```

Comment:
Oops, I didn't realize that we were actually munging the names before passing them to OpenGL... I'm not sure if it'd be more helpful to panic with the name of the munged or un-munged name here.